### PR TITLE
Refactor x86 conv2d_NCHWc

### PR DESCRIPTION
**Pull request by kevinthesun**

### CODE REVIEW

File: topi/python/topi/x86/conv2d.py [hunk 1]
```diff
@@ -553,25 +140,71 @@ def traverse(op):
...
+        kernel[occ * oc_bn + ocb,
+                icc * ic_bn + icb, k_h, k_w],
+        name="kernel_vec")
+
+    return data, kernel
+
 @autotvm.register_topi_compute2("conv2d_NCHWc.x86")
 def conv2d_NCHWc(cfg, data, kernel, strides, padding, dilation, layout, out_layout, out_dtype):
     print('inside x86.conv2d_NCHWC')
     print(cfg)
```
Comment (icemelon) [inline topi/python/topi/x86/conv2d.py]: remove these two lines?

File: python/tvm/relay/op/strategy/generic.py [hunk 2]
```diff
@@ -149,6 +202,32 @@ def conv2d_NCHWc_strategy(attrs, inputs, out_type, target):
...
+        out_layout = attrs.get_str("out_layout")
+        out_dtype = attrs.get_str("out_dtype")
+        out_dtype = inputs[0].dtype if out_dtype == "" else out_dtype
+        return [topi_func(inputs[0], inputs[1], strides, padding, dilation,
+                          data_layout, out_layout, out_dtype)]
+    return compute
+
+@override_native_generic_func("depthwise_conv2d_NCHWc_strategy")
+def depthwise_conv2d_NCHWc_strategy(attrs, inputs, out_type, target):
+    print('inside generic depthwise_conv2d_NCHWc_strategy')
```
Comment (icemelon) [inline python/tvm/relay/op/strategy/generic.py]: remove print & add doc string

File: python/tvm/relay/op/strategy/generic.py [hunk 3]
```diff
@@ -149,6 +202,32 @@ def conv2d_NCHWc_strategy(attrs, inputs, out_type, target):
             wrap_topi_schedule(topi.generic.schedule_conv2d_NCHWc))
     return strategy
 
+# depthwise_conv2d_NCHWc
+def wrap_compute_depthwise_conv2d_NCHWc(topi_func):
```
Comment (icemelon) [inline python/tvm/relay/op/strategy/generic.py:214]: add doc string

File: python/tvm/relay/op/strategy/x86.py [hunk 4]
```diff
@@ -82,6 +120,15 @@ def conv2d_NCHWc_strategy_cpu(attrs, inputs, out_type, target):
                            wrap_topi_schedule(topi.x86.schedule_conv2d_NCHWc))
     return strategy
 
+@depthwise_conv2d_NCHWc_strategy.register("cpu")
+def depthwise_conv2d_NCHWc_strategy(attrs, inputs, out_type, target):
+    print('inside x86 depthwise_conv2d_NCHWc_strategy')
```
Comment (icemelon) [inline python/tvm/relay/op/strategy/x86.py]: remove the print

File: python/tvm/relay/op/strategy/generic.py [hunk 5]
```diff
@@ -149,6 +202,32 @@ def conv2d_NCHWc_strategy(attrs, inputs, out_type, target):
...
+        return [topi_func(inputs[0], inputs[1], strides, padding, dilation,
+                          data_layout, out_layout, out_dtype)]
+    return compute
+
+@override_native_generic_func("depthwise_conv2d_NCHWc_strategy")
+def depthwise_conv2d_NCHWc_strategy(attrs, inputs, out_type, target):
+    print('inside generic depthwise_conv2d_NCHWc_strategy')
+    strategy = _op.OpStrategy()
+    if out_type == "int8":
+        pass
```
Comment (icemelon) [inline python/tvm/relay/op/strategy/generic.py:233]: is this a todo?

Comment (kevinthesun) [reply python/tvm/relay/op/strategy/generic.py:233]: Yes. I'll add it today.

File: topi/python/topi/x86/conv2d.py [hunk 6]
```diff
@@ -365,131 +73,31 @@ def _conv2d_infer_layout(workload, cfg):
...
-#     traverse(outs[0].op)
-#     return s
-
-
-@autotvm.register_topi_compute2("conv2d_nchw.x86")
+# conv2d_nchw compute and schedule is for testing purpose only.
+# It is not used as x86 conv2d NCHW strategy, since
+# alter_op_layout pass and conv2d_NCHWc strategy can
+# eliminate input and output data layout transformations.
 def conv2d_nchw(cfg, data, kernel, strides, padding, dilation, layout, out_dtype):
```
Comment (icemelon) [inline topi/python/topi/x86/conv2d.py:81]: You need to have @autotvm.register_topi_compute2 to allow `cfg` in the arg list

File: topi/python/topi/x86/conv2d.py [hunk 7]
```diff
@@ -365,131 +73,31 @@ def _conv2d_infer_layout(workload, cfg):
...
+                    .astype(out_dtype),
+                    name='output_unpack',
+                    tag='conv2d_nchw')
 
-        scheduled_ops.append(op)
+    return unpacked_out
 
-    traverse(outs[0].op)
-    return s
+def schedule_conv2d_nchw(cfg, outs):
```
Comment (icemelon) [inline topi/python/topi/x86/conv2d.py:101]: similar here, you need to register autotvm to allow cfg.

File: topi/python/topi/x86/conv2d.py [hunk 8]
```diff
@@ -500,7 +108,7 @@ def conv2d_nhwc(_, data, kernel, strides, padding, dilation, layout, out_dtype):
     return nn.conv2d_nhwc(data, kernel, strides, padding, dilation, out_dtype)
 
 @autotvm.register_topi_schedule2("conv2d_nhwc.x86")
-def schedule_conv2d_nhwc2(outs):
+def schedule_conv2d_nhwc(outs):
```
Comment (icemelon) [inline topi/python/topi/x86/conv2d.py]:
```suggestion
def schedule_conv2d_nhwc(cfg, outs):
```

**Pull request merged**